jcommune-service/src/main/java/org/jtalks/jcommune/service/transactional/TransactionalGroupService.java

```diff
@@ -130,7 +134,48 @@ public void saveGroup(Group group) {
      * {@inheritDoc}
      */
     @Override
+    public void saveGroup(GroupAdministrationDto dto) throws NotFoundException {
+        throwValidationExceptionIfNotUnique(dto);
+        Group group = dto.getId() != null ? dao.get(dto.getId()) : new Group();
+        if (group == null) {
+            throw new NotFoundException();
+        }
+        if (!isGroupEditable(group.getName())) {
+            throw new OperationIsNotAllowed("Group is not editable");
+        }
+        group.setName(dto.getName());
+        group.setDescription(dto.getDescription());
+        dao.saveOrUpdate(group);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
     public List<GroupAdministrationDto> getGroupNamesWithCountOfUsers() {
-        return dao.getGroupNamesWithCountOfUsers();
+        List<GroupAdministrationDto> groupNamesWithCountOfUsers = dao.getGroupNamesWithCountOfUsers();
+        setEditableFlag(groupNamesWithCountOfUsers);
+        return groupNamesWithCountOfUsers;
+    }
+
+    private void setEditableFlag(List<GroupAdministrationDto> groups) {
+        for (GroupAdministrationDto dto : groups) {
+            dto.setEditable(isGroupEditable(dto.getName()));
+        }
+    }
+
+    private boolean isGroupEditable(String groupName) {
+        return !AdministrationGroup.isAdministrationGroup(groupName);
+    }
+
+    private void throwValidationExceptionIfNotUnique(GroupAdministrationDto dto) {
+        if (!isUnique(dto)) {
+            throw new ValidationException(asSet(new ValidationError("name", "group.already_exists")));
+        }
+    }
+
+    private boolean isUnique(GroupAdministrationDto dto) {
```

Comment:
I'm more or less ok, just wanted to double-check that a separate method is intended? You could've inline those 2 lines of code in the asserting method.